dbt/adapters/snowflake/relation.py

```diff
@@ -33,3 +51,48 @@ def is_dynamic_table(self) -> bool:
     @classproperty
     def DynamicTable(cls) -> str:
         return str(SnowflakeRelationType.DynamicTable)
+
+    @classproperty
+    def get_relation_type(cls) -> Type[SnowflakeRelationType]:
+        return SnowflakeRelationType
+
+    @classmethod
+    def from_runtime_config(cls, runtime_config: RuntimeConfigObject) -> RelationConfigBase:
+        model_node: ModelNode = runtime_config.model
+        relation_type = SnowflakeRelationType(model_node.config.materialized)
+
+        if relation_config := cls.relation_configs.get(relation_type):
+            return relation_config.from_model_node(model_node)
+
+        raise DbtRuntimeError(
+            f"from_runtime_config() is not supported for the provided relation type: {relation_type}"
+        )
+
+    @classmethod
+    def dynamic_table_config_changeset(
+        cls,
+        dynamic_table: SnowflakeDynamicTableConfig,
```

Comment:
So this is pointing to what we're creating? Should this be like `target_` prefixed or something like that to highlight this is the to-be-built copy of an existing table?

Reply:
I'm definitely focusing on the new snapshot, which lives in the user config, as the "primary" view of the dynamic table. I did that because that's where I am when I'm running this, in the future looking back at the past. However, as noted above, this should probably change. I also don't like `relation_results` since it's very vague. This was the result of several back and forth's in reviews. I think I just might bite the bullet, and do the parsing for the `existing_dynamic_table` outside of this method as well, and then wind up with `existing_dynamic_table` and `new_dynamic_table` as the arguments, which is more clear.

Reply:
I'd love a `﻿﻿new_dynamic_table` name everywhere personally!